src/strands/experimental/bidirectional_streaming/tests/test_bidi_openai.py

```
@@ -0,0 +1,287 @@
+                audio_data = event["audioOutput"]["audioData"]
+                
+                if not context.get("interrupted", False):
+                    await context["audio_out"].put(audio_data)
+            
+            # Handle text output (transcripts)
+            elif "textOutput" in event:
+                text_output = event["textOutput"]
+                role = text_output.get("role", "assistant")
+                text = text_output.get("text", "").strip()
+                
+                if text:
+                    if role == "user":
+                        print(f"User: {text}")
+                    elif role == "assistant":
+                        print(f"Assistant: {text}")
+            
+            # Handle interruption detection
+            elif "interruptionDetected" in event:
+                context["interrupted"] = True
+            
+            # Handle connection events
+            elif "BidirectionalConnectionStart" in event:
+                pass  # Silent connection start
+            elif "BidirectionalConnectionEnd" in event:
+                context["active"] = False
+                break
+    
+    except asyncio.CancelledError:
+        pass
+    except Exception as e:
+        print(f"Receive handler error: {e}")
+    finally:
+        pass
+
+
+async def send(agent, context):
+    """Send audio from microphone to agent."""
+    try:
+        while context["active"]:
+            try:
+                audio_bytes = await asyncio.wait_for(context["audio_in"].get(), timeout=0.1)
+                
+                # Create audio event in expected format
+                audio_event = {
+                    "audioData": audio_bytes,
+                    "format": "pcm",
+                    "sampleRate": 24000,
+                    "channels": 1
+                }
+                
+                await agent.send(audio_event)
+                
+            except asyncio.TimeoutError:
```

We could discuss tomorrow when we want to fail hard, when we want to handle it silently